### PR TITLE
[CI] Add CodeQL analysis

### PR DESCRIPTION
**Pull request by foundrytom**
Part of #21. This is the stock GitHub template, generated by their onboarding wizard.

Mirror PR [here](https://github.com/foundrytom/OpenAssetIO/pull/3) to see it in use.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (feltech) [inline .github/workflows/codeql-analysis.yml:1]: [consider][minor] I think we could remove this preamble, it doesn't add much info other than to make clear this was auto generated.

**Review by feltech — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
+
+    - name: Perform CodeQL Analysis
```
[the remark this answers is not in the document]
Comment (foundrytom) [reply .github/workflows/codeql-analysis.yml:69]: That was my assumption too

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (foundrytom) [reply .github/workflows/codeql-analysis.yml]: Improved [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/92/commits/f3abe1e1a49fd7c4e8cc8045d027e3e839e9522a).

**Comment by feltech** [comment]
LGTM squash and merge when ready

**Pull request merged**